src/SocketContext.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import io from 'socket.io-client';
+import { useAuth } from './AuthProvider';
+import NotificationModal from '../src/components/common/notificationModal';
+
+const SocketContext = createContext();
+
+const setupSocketListeners = (socketIo, userId, setNotificationMessage, setIsModalOpen) => {
+  const handleNotification = (data) => {
+    console.log('Received notification:', data.content);
+    setNotificationMessage(data.content);
+    setIsModalOpen(true);
+  };
+
+  socketIo.on(`applications-${userId}`, handleNotification);
+  socketIo.on(`approveMessage-${userId}`, handleNotification);
+  socketIo.on(`rejectMessage-${userId}`, handleNotification);
+
+  socketIo.on('disconnect', () => {
+    console.log('Disconnected');
+  });
+};
+
+export const SocketProvider = ({ children }) => {
+  const { userData } = useAuth();
+  const [socket, setSocket] = useState(null);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [notificationMessage, setNotificationMessage] = useState('');
+
+  useEffect(() => {
+    if (!userData) {
+      console.error('No user data found');
+      return;
+    }
+
+    const userId = userData._id;
+
+    const socketIo = io('http://localhost:8000', {
+      query: { userId: userId },
+    });
+
+    socketIo.on('connect', () => {
+      console.log('Connected with socket ID:', socketIo.id);
```

Comment:
small suggestion to change intended console.logs into console.info. It will help us to be sure that we need to save this console message (we will throw the baby out with the bathwater)

Reply:
Changed)